components/SwipeListView.js

```diff
@@ -38,15 +38,26 @@ class SwipeListView extends Component {
 		if (this.props.closeOnRowBeginSwipe && this.openCellId && this.openCellId !== id) {
 			this.safeCloseOpenRow();
 		}
+
+    if (this.props.swipeGestureBegan) {
+      this.props.swipeGestureBegan(id);
+    }
+
 	}
 
-	onRowOpen(secId, rowId, rowMap) {
+	onRowOpen(rowData, secId, rowId, rowMap) {
 		const cellIdentifier = `${secId}${rowId}`;
 		if (this.openCellId && this.openCellId !== cellIdentifier) {
 			this.safeCloseOpenRow();
 		}
 		this.openCellId = cellIdentifier;
-		this.props.onRowOpen && this.props.onRowOpen(secId, rowId, rowMap);
+		this.props.onRowOpen && this.props.onRowOpen(rowData, secId, rowId, rowMap);
```

Comment:
Hm, introducing `rowData` here and into all of the other callbacks as the first argument is going to be a big breaking change that I don't feel is necessary. If anything it should be the 4th argument.

But do we need to pass `rowData` at all? You already have the `rowMap` so you can access any row you need. And the parent component has to pass `rowData` to the `SwipeListView` so why do we need to pass it back to the parent in the callback?

I'm not sure I see the use case here

Reply:
To be honest, I don't remember why I had it. I do find it annoying that I don't have access to the data being passed to a row object.  Anyway, I'm OK to get rid of this.